cdap-common/src/main/java/co/cask/cdap/common/conf/Constants.java

```diff
@@ -266,6 +268,8 @@
       /** for the address (hostname) of the dataset server. */
       public static final String ADDRESS = "dataset.service.bind.address";
       public static final String PORT = "dataset.service.bind.port";
+      public static final String ANNOUNCE_ADDRESS = "dataset.service.announce.address";
```

Comment:
I agree with the bind address as well. It doesn't really make sense. Have two configs means it's easy for the user to miss one.